quinn-proto/src/config.rs

```diff
@@ -985,3 +991,18 @@ impl fmt::Debug for IdleTimeout {
         self.0.fmt(f)
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct AcceptConfig {
```

Comment:
Is there a use case for giving different preferred addresses to different incoming connections at the same time? If not, it'd be simpler to make this a field in `ServerConfig`. Note that `ServerConfig`s can be swapped out live.

Reply:
my use case is only giving a preferred address to a percentage of incoming requests and then measuring the relative performance of those connections.

Reply:
Makes sense. Let's nonetheless make these options on `ServerConfig`, but modify `accept` to allow a custom `ServerConfig` to be specified. That way we expose a single path for per-connection customization of all per-connection settings.

Reply:
Assuming you mean `TransportConfig`, is it guaranteed that everything on `TransportConfig` will always be able to be applied during `accept`? If so, should we just remove ServerConfig.transport_config and put it on accept, so that there's only one place for it?

Reply:
> Assuming you mean TransportConfig

No, I mean `ServerConfig`. `TransportConfig` contains parameters that are relevant to both incoming and outgoing connections. `ServerConfig` includes parameters that are only relevant to incoming connections. Preferred addresses are only relevant to incoming connections.

> is it guaranteed that everything on TransportConfig will always be able to be applied during accept?

`ServerConfig` is largely per-connection. Varying your initial or retry token keys between connections on a single endpoint might cause undesired behavior, but otherwise you can mess with it pretty freely.

Reply:
ServerConfig contains TransportConfig so I'm not sure how the categorization you just said shakes out, but in particular I don't think overriding `crypto` on a per-accept basis works because it is used before `Incoming` is emitted. Until someone more familiar with the library could refactor out the proper connection-level config here it seems to be like an AcceptConfig is cleanest.

Reply:
> I don't think overriding crypto on a per-accept basis works

Not only does it work, it's specifically desirable: we might want to present different certificates to different clients, for example. The pre-connection uses are limited to the cases I described above. Even those cases could likely be factored out with more work, but that is out of scope of this PR.

Please use `ServerConfig` as described.